File: 242ValidAnagram.cpp
#include<iostream>
#include<string>
#include<vector>
#include<unordered_map>
#include<algorithm>
using namespace std;

class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map <char, int> smap;
        unordered_map <char, int> tmap;

        if (s.size() != t.size()) {
            return false;
        }
        for (int i = 0; i < s.size(); i++) {
            smap[s[i]]++;
            tmap[t[i]]++;
        }

        for (int i = 0; i < s.size(); i++) {
            if (smap[i] != tmap[i]) {
                return false;
            } 
        }
        return true;
        
    }
};

int main () {
    Solution S;
    string s = "anagram";
    string t = "nagaram";

    cout << S.isAnagram(s, t) << endl;
}

File: 242ValidAnagram.py
class Solution(object):
    def isAnagram(self, s, t):
        myMap = {}
        
        for i in s:
            if i not in myMap:
                myMap[i] = 0
            myMap[i] += 1
        
        for i in t:
            if i not in myMap:
                myMap[i] = 0
            myMap[i] -= 1

        for val in myMap.values():
            if val != 0:
                return False
        return True

    class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        sortedS = sorted(s)
        sortedT = sorted(t)

        if sortedS == sortedT:
            return True
        return False

        

if __name__ == "__main__":
    S = Solution()
    s = "anagram"
    t = "nagaram"
    print(S.isAnagram(s,t))
